apps/src/code-studio/components/pairing/StudentSelector.jsx

```diff
@@ -47,29 +69,48 @@ export default class StudentSelector extends React.Component {
     } else if (this.props.students.length === 0) {
       return <span>{i18n.noStudentsInSection()}</span>;
     }
+    const {students} = this.props;
+    const {selectedStudentIds} = this.state;
+    const exceededMaximum = selectedStudentIds.length >= 4;
 
-    const studentDivs = this.props.students.map(student => {
-      let className = 'student selectable';
-      if (this.state.selectedStudentIds.indexOf(student.id) !== -1) {
-        className = 'student selectable selected';
+    const studentBtns = students.map(student => {
+      let className = 'selected';
+      let disabled = false;
+      if (selectedStudentIds.indexOf(student.id) === -1) {
+        className = 'selectable';
+        disabled = exceededMaximum;
       }
+      let buttonStyle = styles.enabled;
```

Comment:
NIT: (again, only change if you want) It feels a bit odd to have styles.enabled later get merged with styles.disabled. Would it make more sense to name this something like styles.pairButton or styles.button?